src/main/java/org/geppetto/simulator/remote/services/NeuronNSGSimulatorService.java

```diff
@@ -0,0 +1,320 @@
+import org.geppetto.simulator.external.services.AExternalProcessNeuronalSimulator;
+import org.ngbw.directclient.CiCipresException;
+import org.ngbw.directclient.CiClient;
+import org.ngbw.directclient.CiJob;
+import org.ngbw.restdatatypes.ErrorData;
+import org.ngbw.restdatatypes.LimitStatus;
+import org.ngbw.restdatatypes.ParamError;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+/**
+ * Wrapper class for NEURON at Neuroscience Gateway
+ * 
+ * @author Adrian Quintana (adrianquintana@gmail.com)
+ *
+ */
+@Service
+public class NeuronNSGSimulatorService extends AExternalProcessNeuronalSimulator
+{
+
+	protected Path filePath = null;
+
+	private static Log logger = LogFactory.getLog(NeuronNSGSimulatorService.class);
+
+	@Autowired
+	private SimulatorConfig neuronSimulatorConfig;
+
+	@Autowired
+	private RemoteSimulatorConfig neuronNSGExternalSimulatorConfig;
+
+	private CiClient myClient;
+	private CiJob jobStatus;
+
+	@Override
+	public void initialize(DomainModel model, IAspectConfiguration aspectConfiguration, ExperimentState experimentState, ISimulatorCallbackListener listener, GeppettoModelAccess modelAccess)
+			throws GeppettoInitializationException, GeppettoExecutionException
+	{
+		super.initialize(model, aspectConfiguration, experimentState, listener, modelAccess);
+
+		if(model instanceof ExternalDomainModel)
+		{
+			originalFileName = (String) model.getDomainModel();
+		}
+		else
+		{
+			throw new GeppettoExecutionException("Unexpected domain model inside NEURON NSG Simulator service");
+		}
+
+		try
+		{
+			// Set directory to execute from and output folder
+			File originalFilePath = new File(originalFileName);
+			directoryToExecuteFrom = originalFilePath.getParentFile().getAbsolutePath();
+			outputFolder = directoryToExecuteFrom;
+
+			// Rename main script to input.py (NSG requirement)
+			File renamedfilePath = new File(directoryToExecuteFrom + "/input.py");
+			Files.copy(originalFilePath.toPath(), renamedfilePath.toPath(), StandardCopyOption.REPLACE_EXISTING);
+
+			// Create Results Folder (it is needed because of the way the neuron code is generated in the export library)
+			File resultsFolder = new File(directoryToExecuteFrom + "/results");
+			resultsFolder.mkdir();
+			File emptyFile = new File(resultsFolder, "empty");
+			emptyFile.createNewFile();
+
+			// Zip the folder creating an inner folder input
+			Zipper zipper = new Zipper(originalFilePath.getParentFile().getParentFile().getAbsolutePath() + "/input.zip", "input");
+			filePath = zipper.getZipFromDirectory(new File(directoryToExecuteFrom));
+
+			//Initialise client for remote execution
+			if (neuronNSGExternalSimulatorConfig.getSimulatorParameters().containsKey("appName")){
+				Map<String, String> endUserHeaders = new HashMap<String, String>();
+				endUserHeaders.put("cipres-eu", "AdrianQuintana");
+				endUserHeaders.put("cipres-eu-email", "AdrianQuintana@gmail.com");
```

Comment:
@tarelli I have just committed some code so that these headers are extracted from the app-config file. For the moment, we are going to be using the same contact details for every user. In the next release I would like to add the user email to the user table and use it here. Nothing to worry about for the moment!